internal/config/defaults_test.go

```diff
@@ -0,0 +1,130 @@
+package config
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+
+	"gopkg.in/yaml.v3"
+)
+
+func TestDefaultCategoriesAndLanguagesConfig(t *testing.T) {
+	tests := []struct {
+		name       string
+		configData interface{}
+		configType string
+		expected   JioTVConfig
+	}{
+		{
+			name: "JSON config with default categories and languages",
+			configData: map[string]interface{}{
+				"default_categories": []int{1, 2, 3},
+				"default_languages":  []int{6, 1},
+				"debug":              true,
+			},
+			configType: "json",
+			expected: JioTVConfig{
+				DefaultCategories: []int{1, 2, 3},
+				DefaultLanguages:  []int{6, 1},
+				Debug:             true,
+			},
+		},
+		{
+			name: "YAML config with default categories and languages",
+			configData: map[string]interface{}{
+				"default_categories": []int{8, 5},
+				"default_languages":  []int{1},
+				"epg":                false,
+			},
+			configType: "yaml",
+			expected: JioTVConfig{
+				DefaultCategories: []int{8, 5},
+				DefaultLanguages:  []int{1},
+				EPG:               false,
+			},
+		},
+		{
+			name: "Empty arrays should work",
+			configData: map[string]interface{}{
+				"default_categories": []int{},
+				"default_languages":  []int{},
+				"title":              "Test App",
+			},
+			configType: "json",
+			expected: JioTVConfig{
+				DefaultCategories: []int{},
+				DefaultLanguages:  []int{},
+				Title:             "Test App",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create temporary config file
+			tmpFile, err := os.CreateTemp("", "test-config-*."+tt.configType)
+			if err != nil {
+				t.Fatalf("Failed to create temp file: %v", err)
+			}
+			defer os.Remove(tmpFile.Name())
+
+			// Write config data to file
+			var data []byte
+			if tt.configType == "json" {
+				data, err = json.Marshal(tt.configData)
+			} else if tt.configType == "yaml" {
+				data, err = yaml.Marshal(tt.configData)
+			}
```

Comment:
For selecting behavior based on a string value like `configType`, a `switch` statement is more idiomatic and often more readable than an `if-else if` chain. It also makes it easier to add more config types in the future.

```suggestion
var data []byte
	switch tt.configType {
	case "json":
		data, err = json.Marshal(tt.configData)
	case "yaml":
		data, err = yaml.Marshal(tt.configData)
	default:
		t.Fatalf("unsupported config type: %s", tt.configType)
	}
```